HACKING.md

```diff
@@ -1,10 +1,88 @@
-The `manifests/` directory contains OpenShift manifests that will set up a Jenkins pipeline
-using the [Kubernetes Jenkins plugin](https://github.com/jenkinsci/kubernetes-plugin).
+The `manifests/pipeline.yaml` file contains an OpenShift template that
+will set up a Jenkins pipeline using the
+[Kubernetes Jenkins plugin](https://github.com/jenkinsci/kubernetes-plugin).
 
-### Create a Jenkins instance with a persistent volume backing store:
+These instructions will cover both local developer workflows, as well as
+interacting with the production pipeline in CentOS CI. A section header
+may indicate that the section only applies to one of the two workflows.
 
+### [LOCAL] Set up an OpenShift cluster
+
+First, make sure to install `oci-kvm-hook` on your host system (not in a
+pet container). This is required to ensure that the pipeline has access
+to `/dev/kvm`:
+
+```
+rpm-ostree install oci-kvm-hook # if on Silverblue
+dnf install -y oci-kvm-hook # if on traditional
+```
+
+We will use `oc cluster up` to set up a local cluster for testing. To do
+this, simply obtain the OpenShift binary from
+[the latest release](https://github.com/openshift/origin/releases/latest)
+(though note that I'm currently still using the binaries from v3.9, so
+pick that release if you hit issues).
+
+And now, bring up a v3.6.1 cluster (we want to match the CentOS CI
+version):
+
+```
+oc cluster up --version v3.6.1
+```
+
+To have persistent configs and data, I would recommend specifying the
+dirs to use.
+
+```
+oc cluster up --version v3.6.1 \
+    --host-config-dir=/srv/oc/config \
+    --host-data-dir=/srv/oc/data \
+    --host-pv-dir=/srv/oc/pv
+```
+
+Then when doing `oc cluster up` in the future, you can use
+`--use-existing-config`.
```

Comment:
can you provide a full example of the command to run with `--use-existing-config` ? Is it just `oc cluster up --use-existing-config` ?